docs/src/modules/ROOT/partials/concepts-intro.adoc

```diff
@@ -44,6 +44,60 @@ You create a continuous incoming stream of events with the xref:java:http-endpoi
+
+Akka components are marked with a `@ComponentId` or `@HttpEndpoint` annotation to identify them to the runtime.
+
+=== Agents
+
+image:ROOT:agent.png[Agent,width=100,float=left]An _Agent_ interacts with an AI model to perform a specific task. It is typically backed by a large language model (LLM). It maintains contextual history in a session memory, which may be shared between multiple agents that are collaborating on the same goal. It may provide function tools and call them as requested by the model. For more information see xref:java:agents.adoc[Agents].
+
+=== Endpoints
+
+image:ROOT:endpoint.png[Endpoints,width=100,float=left] _Endpoints_ are defined points of interaction for services that allow external clients to communicate via HTTP or gRPC. They facilitate the integration and communication between the other types of internal Akka components. Unlike other Akka components, endpoints will live in your `api` package. For more information see xref:java:http-endpoints.adoc[HTTP Endpoints] and xref:java:grpc-endpoints.adoc[gRPC Endpoints].
+
+=== Workflows
+
+image:ROOT:workflow.png[Workflows,width=100,float=left] _Workflows_ enable the developer to implement long-running, multi-step business processes while focusing exclusively on domain and business logic. Technical concerns such as delivery guarantees, scaling, error handling and recovery are managed by Akka. For more information see xref:java:workflows.adoc[Workflows].
+
+=== Entities
+
+image:ROOT:entity.png[Entities,width=100,float=left] _Entities_ are the core components of Akka and provide persistence and state management. They map to your https://martinfowler.com/bliki/DDD_Aggregate.html[_domain aggregates_, window="new"]. If you have a "Customer" domain aggregate, you almost certainly will have a `CustomerEntity` component to expose and manipulate it. This separation of concerns allows the domain object to remain purely business logic focused while the Entity handles runtime mechanics. Additionally, you may have other domain objects that are leafs of the domain aggregate. These do not need their own entity if they are just a leaf of the aggregate. An address is a good example.
+
+There are two types of entities in Akka. Their difference lies in how they internally function and are persisted.
+
+==== Key Value Entities
+
+image:ROOT:key-value-entity.png[Key Value Entities,width=100,float=left] _Key Value Entities_ are, as the name implies, an object that is stored and retrieved based on a key - an identifier of some sort. The value is the entire state of the object. Every write to a Key Value Entity persists the entire state of the object. Key Value Entities are similar in some ways to database records. They write and effectively lock the whole row. They still use an underlying event-based architecture so other components can subscribe to the stream of their updates. For more information see xref:java:key-value-entities.adoc[Key Value Entities].
+
+==== Event Sourced Entities
```

Comment:
It might make sense to introduce event sourced entities first, since key value entities require some knowledge of them.